src/main/java/com/example/repository/BookRepositoryImpl.java

```diff
@@ -0,0 +1,28 @@
+package com.example.repository;
+
+import com.example.model.Book;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+@Repository
+@RequiredArgsConstructor
+public class BookRepositoryImpl implements BookRepository {
+    @PersistenceContext
+    private EntityManager entityManager;
+
+    @Transactional
+    @Override
+    public Book save(Book book) {
+        entityManager.persist(book);
+        return book;
+    }
```

Comment:
here we are supposed to use explicit transaction management approach using Hibernate's Session and Transaction